elastic/kibana
# [SecuritySolution] Retrieve onboarding steps

## Summary

Fixes https://github.com/elastic/kibana/issues/181190
Relevant PRs: https://github.com/elastic/kibana/pull/143598 | https://github.com/elastic/kibana/pull/163739


### Steps to Verify: 
1. Entering [these](https://p.elstc.co/paste/sEmk++Tb#mjwuX7IN8hIN+kOy5gdtfweQNi9sUl+4lVRAewc6hR+) in your kibana.dev.yml
2. Execute this command to set the Guided onboarding steps to alertsCases
```
curl --location --request PUT 'http://localhost:5601/internal/guided_onboarding/state' \
--header 'kbn-xsrf: cypress-creds' \
--header 'x-elastic-internal-origin: security-solution' \
--header 'Content-Type: application/json' \
--header 'Authorization: Basic {YOUR_AUTH_TOKEN}' \.    // If you are using Postman just fill in the Auth tab
--data '{
  "status": "in_progress",
  "guide": {
    "isActive": true,
    "status": "in_progress",
    "steps": [
      { "id": "add_data", "status": "complete" },
      { "id": "rules", "status": "complete" },
      { "id": "alertsCases", "status": "active" }
    ],
    "guideId": "siem"
  }
}'
```
3. Make sure you have alerts available.
4. To test the old flyout with Guided onboarding tour, please go to Stack Management > Advanced settings > Expandable flyout **OFF**

### It compatible with the new expandable flyout:
1. It shows `expandable flyout tour` when the guided onboarding tour is **not enabled**.
3. The first two steps should be `hidden` when the `left` expandable is opened.
5. Most of the guided onboarding tour steps should be hidden when `Add to new case` flyout or `Add to existing case` modal opened.
6. Once the test case is created, the `insight section` and `correlation tab`  should be opened automatically to fetch cases.
7. `expandable flyout tour` should be visible again after the guided onboarding tour is finished.





https://github.com/elastic/kibana/assets/6295984/b19bfce9-ec02-4291-b616-e24d3e984a03






### It compatible with the old flyout: 



https://github.com/elastic/kibana/assets/6295984/b10b8bdf-e159-4663-b455-1f4541358a11


### Checklist

Delete any items that are not applicable to this PR.
